### PR TITLE
Run balloon test during migration

### PR DESCRIPTION
**Pull request by suqinhuang**
Keep balloon testing until migration finished

Signed-off-by: Suqin Huang <shuang@redhat.com>

id: 1427014

### CODE REVIEW

File: qemu/tests/balloon_during_migration.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    bg.start()
+
+    try:
+        while bg.isAlive():
+            balloon_test.balloon_memory(int(random.uniform(min_sz, max_sz)))
+    except Exception:
+        bg.join()
+        test.fail("Balloon test failed")
+    finally:
+        session.close()
```
Comment (aliang123) [inline qemu/tests/balloon_during_migration.py]: To check dst vm's balloon status after migration will be better.

File: qemu/tests/balloon_during_migration.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    login_timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=login_timeout)
+
+    mig_timeout = float(params.get("mig_timeout", "3600"))
+    mig_protocol = params.get("migration_protocol", "tcp")
+    mig_cancel_delay = int(params.get("mig_cancel") == "yes") * 2
+
+    balloon_test = BallooningTestWin(test, params, env)
+    min_sz, max_sz = balloon_test.get_memory_boundary()
+    bg = utils_misc.InterruptedThread(vm.migrate, (mig_timeout,
```
Comment (aliang123) [inline qemu/tests/balloon_during_migration.py]: I think no need to use param "mig_cancel_delay" here.

**Comment by suqinhuang** [comment]
@aliang123 updated

**Comment by CongLi** [comment]
@lijinlijin @aliang123 @yumingfei 

Could you please help review this PR?

**Comment by lijinlijin** [comment]
smp_8.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.balloon_in_use.with_live_migration.during_bg_test: PASS (323.44 s)

**Comment by lijinlijin** [comment]
smp_8.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.balloon_in_use.with_live_migration.after_bg_test: - ERROR (4089.65 s)

06:34:52 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_framework-54.1-py2.7.egg/avocado/core/test.py:630
06:34:52 ERROR| Traceback (most recent call last):
06:34:52 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-54.0-py2.7.egg/avocado_vt/test.py", line 271, in runTest
06:34:52 ERROR|     raise self.__status  # pylint: disable=E0702
06:34:52 ERROR| VMMigrateTimeoutError: Timeout expired while waiting for migration to finish

**Comment by lijinlijin** [comment]
Run balloon_in_use.with_live_migration.during_bg_test two more times,all failed as following:
21:26:47 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-54.0-py2.7.egg/avocado_vt/test.py:245
21:26:47 ERROR| Traceback (most recent call last):
21:26:47 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-54.0-py2.7.egg/avocado_vt/test.py", line 397, in _runTest
21:26:47 ERROR|     run_func(self, params, env)
21:26:47 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-54.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
21:26:47 ERROR|     return fn(*args, **kwargs)
21:26:47 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/balloon_during_migration.py", line 38, in run
21:26:47 ERROR|     test.fail("Balloon test failed: %s" % err)
21:26:47 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_framework-54.1-py2.7.egg/avocado/core/test.py", line 796, in fail
21:26:47 ERROR|     raise exceptions.TestFail(message)
21:26:47 ERROR| TestFail: Balloon test failed: Balloon memory fail with error message: Could not receive data from monitor    ([Errno 104] Connection reset by peer)    [context: Change VM memory to 3794]

After migration finish,the last balloon_memory in src guest fail.

**Pull request closed without merging**